### PR TITLE
docs: update with-ionic-angular.mdx

### PR DESCRIPTION
**Pull request by bobbyg603**
## What kind of change does this PR introduce?

Docs update

## What is the current behavior?

The samples from the ionic docs are outdated and don't compile in the latest versions of ionic/Angular.

## What is the new behavior?

The samples compile and runs. Additionally, the email input on the account page was blank so I fixed that as well.

## Additional context

* Fix compile errors
* Update syntax to match @supabase/supabase-js v2
* Correct spelling/grammar mistakes found by Grammarly

I waffled a bit on what to do about the untyped errors. I opted for any because the point of this tutorial is a minimal demonstration of supabase that's approachable to a beginner/intermediate level developer. With that in mind, it felt more important to keep the code concise instead of type-checking errors in every catch block.  I'm open to feedback here.

Additionally, I was indecisive about handling the API change regarding returning errors instead of throwing them. I opted to throw the error immediately if one was returned because that was the only way I could guarantee that the sample snippets worked the same as before this update without doing a deep dive. I'm open to feedback here as well.

### CODE REVIEW

File: apps/docs/pages/guides/getting-started/tutorials/with-ionic-angular.mdx [hunk 1]
```diff
@@ -75,43 +75,47 @@ export class SupabaseService {
...
   get user() {
-    return this.supabase.auth.user()
+    return this.supabase.auth.getUser()
+      .then(({ data }) => data?.user)
   }
 
   get session() {
-    return this.supabase.auth.session()
+    return this.supabase.auth.getSession()
+      .then(({ data}) => data?.session)
```
Comment (bobbyg603) [inline apps/docs/pages/guides/getting-started/tutorials/with-ionic-angular.mdx:84]:
```suggestion
      .then(({ data }) => data?.session)
```

File: apps/docs/pages/guides/getting-started/tutorials/with-ionic-angular.mdx
Comment (burggraf) [inline apps/docs/pages/guides/getting-started/tutorials/with-ionic-angular.mdx:1]: One quick change you should make:
change:
`ts environment.ts` to `src/environments/environment.ts`

**Review by burggraf — DISMISSED** [review]
One quick change you should make:
change:
`ts environment.ts` to `src/environments/environment.ts`

File: apps/docs/pages/guides/getting-started/tutorials/with-ionic-angular.mdx [hunk 2]
```diff
@@ -38,7 +38,7 @@ Then let's install the only additional dependency: [supabase-js](https://github.
 npm install @supabase/supabase-js
 ```
 
-And finally we want to save the environment variables in the `environment.ts` file.
+And finally, we want to save the environment variables in the `environment.ts` file.
```
Comment (bobbyg603) [inline apps/docs/pages/guides/getting-started/tutorials/with-ionic-angular.mdx:41]:
```suggestion
And finally, we want to save the environment variables in the `src/environments/environment.ts` file.
```

File: apps/docs/pages/guides/getting-started/tutorials/with-ionic-angular.mdx
Comment (bobbyg603) [reply apps/docs/pages/guides/getting-started/tutorials/with-ionic-angular.mdx:1]: thanks for the review, addressed this in 🙂 [3799e67](https://github.com/supabase/supabase/pull/18613/commits/3799e678dc4e02ac0f74de7f70555a1bfc1c3bb7)

**Comment by bobbyg603** [comment]
hi @burggraf, can you please approve/merge? thank you!

File: apps/docs/pages/guides/getting-started/tutorials/with-ionic-angular.mdx [hunk 3]
```diff
@@ -135,8 +139,8 @@ export class SupabaseService {
 
 ### Set up a Login route
 
-Let's set up an route to manage logins and sign ups. We'll use Magic Links, so users can sign in with their email without using passwords.
-Create an **LoginPage** with `ionic g page login` Ionic CLI command.
+Let's set up a route to manage logins and sign-ups. We'll use Magic Links so users can sign in with their email without using passwords.
```
Comment (charislam) [inline apps/docs/pages/guides/getting-started/tutorials/with-ionic-angular.mdx:142]:
```suggestion
Let's set up a route to manage logins and signups. We'll use Magic Links so users can sign in with their email without using passwords.
```

File: apps/docs/pages/guides/getting-started/tutorials/with-ionic-angular.mdx [hunk 4]
```diff
@@ -135,8 +139,8 @@ export class SupabaseService {
...
+Create a **LoginPage** with `ionic g page login` Ionic CLI command.
```
Comment (charislam) [inline apps/docs/pages/guides/getting-started/tutorials/with-ionic-angular.mdx:143]:
```suggestion
Create a **LoginPage** with the `ionic g page login` Ionic CLI command.
```

File: apps/docs/pages/guides/getting-started/tutorials/with-ionic-angular.mdx [hunk 5]
```diff
@@ -291,7 +307,7 @@ export class AccountPage implements OnInit {
 
 ### Launch!
 
-Now that we have all the components in place, let's update **AppComponent**:
+Now that we have all the components in place let's update **AppComponent**:
```
Comment (charislam) [inline apps/docs/pages/guides/getting-started/tutorials/with-ionic-angular.mdx:310]:
```suggestion
Now that we have all the components in place, let's update **AppComponent**:
```

**Review by charislam — APPROVED** [review]

**Comment by charislam** [comment]
LGTM, thanks for this one!

**Comment by bobbyg603** [comment]
> LGTM, thanks for this one!

Thanks for the review @charislam! I noticed the merge was blocked by the prettier build rule. I ran prettier and pushed a commit. Can you please re-enable auto-merge? Thank you!

**Comment by bobbyg603** [comment]
Thanks for merging this @charislam 🎉!

**Pull request merged**